common/beerocks/tlvf/src/include/beerocks/tlvf/beerocks_message.h

```diff
@@ -27,6 +30,35 @@
 
 namespace beerocks {
 
+class beerocks_header {
+    public:
+        beerocks_header(std::unique_ptr<ieee1905_1::TlvList> contents)
+            : m_contents(std::move(contents))
+        {}
+        beerocks_message::eAction action() { return actionhdr()->action(); }
+        uint8_t action_op() { return actionhdr()->action_op(); };
+        uint16_t id() { return actionhdr()->id(); }
+        template <class T> std::shared_ptr<T> addClass()
+        {
+            return m_contents->addClass<T>();
+        }
+        template <class T> std::shared_ptr<T> getClass()
+        {
+            return m_contents->getClass<T>();
+        }
```

Comment:
Note that the idea was that these would be removed again at the end of the PR, because the users instead use a wrapper, e.g. `addAction()` and `getAction()` that take into account the action header. Or that m_contents is public so callers can use it directly.

Reply:
Done

Reply:
I kept the wrapper functions for backward compatability. Plus I think having some forward wrappers such as this is OK - it also makes lines shorter.

Reply:
There is no backward compatibility for these, because they didn't exist before this PR.

Avoiding the forwarding wrappers is because it makes class instantiation more difficult. But the main point actually is that instead of a simple wrapper, you would typically want forwarding to be done with more intelligence. That's why I was talking about addAction() and getAction().

Reply:
But now that `beerocks_header` extends `TlvList`, it gets `addClass` and `getClass` by default. We can add wrappers now `addAction` and `getAction` but we won't be able to "hide" `addClass` and `getClass`.